### DISCUSSION TITLE
Use connected A2DP phone as audio device input in JUCE

### DISCUSSION THREAD

**Discussion by jeromeDms** [Arkq/bluez-alsa#466] [Q&A]
Dear all
I'm searching for a solution to use a mobile phone connected to Raspberry Pi3 Bluetooth as a virtual audio device input.
I developed a JUCE app that process audio from the device input, apply some effects and send the buffers to the output.
I would now like to replace the physical input with a A2DP mobile phone.
Could bluez-alsa help to achieve this ?
Ideally the default audio input would be configured as the Bluetooth input at 44100Hz.
Any tips would be greatly appreciated 
Thanks

**Comment by borine** [comment]
As this is a general question rather that an issue, I have moved it to the Q&A discussion forum.

If by "device input" you mean an ALSA PCM capture device, then it should be possible. BlueALSA presents Bluetooth A2DP sources to applications as ALSA capture devices. There is lots of information on how to build, install and use BlueALSA on the this project homepage, the wiki, and manual pages.
